packages/docs/src/pages/guides/jsx-pragma.mdx

```diff
@@ -55,3 +64,120 @@ The preferred way of using these functions is by adding the custom pragma commen
 See the [`sx` prop][] docs to learn more.
 
 [`sx` prop]: /sx-prop
+
+## Automatic JSX runtime
+
+React v17 introduced a
+[new JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html),
+also called "automatic runtime" (backported to React v16.14.0) If you use any of
+those versions together with a configurable transpiler (such as Babel or
+TypeScript), you can configure JSX to use the automatic runtime (globally, for
+your entire app), and no longer need to use the custom pragma comments in your
+files to use `sx`.
+
+Here are a few examples showing how to do that depending how you build your
+application:
+
+### Using @babel/preset-react
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      '@babel/preset-react',
+      {
+        importSource: 'theme-ui', // or '@theme-ui/core'
+        runtime: 'automatic',
+        throwIfNamespace: false,
+      },
+    ],
+  ],
+  // ...
+}
+```
+
+NOTE: this requires
+[babel >= 7.9.0](https://babeljs.io/docs/en/babel-preset-react#react-automatic-runtime)
+
+### Using @babel/plugin-transform-react-jsx
+
+This plugin is included in the preset above, but can also used as a standalone
+babel plugin.
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      '@babel/plugin-transform-react-jsx',
+      {
+        importSource: 'theme-ui', // or '@theme-ui/core'
+        runtime: 'automatic',
+        throwIfNamespace: false,
+      },
+    ],
+  ],
+  // ...
+}
+```
+
+NOTE: this requires
+[babel >= 7.9.0](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#react-automatic-runtime)
+
+### Using TypeScript
+
+If you use TypeScript to transpile your source code with `tsc` (or only
+typecheck), or for instance to run tests with `ts-jest`
+
+```json
+// tsconfig.json
+{
+  "compilerOptions": {
+    "jsx": "react-jsx",
+    "jsxImportSource": "theme-ui"
+  }
+  // ...
+}
+```
+
+NOTE: this requires
+[TypeScript >= 4.1](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#react-17-jsx-factories)
+
+NOTE: in order to typecheck `sx` using automatic runtime, you will also need to
+add `sx` manually to React types:
+
+```ts
+// To add `css` if you're using it somewhere
+/// <reference types="@emotion/react/types/css-prop" />
+
+import {} from 'react'
+import { ThemeUIStyleObject } from 'theme-ui' // or '@theme-ui/core'
+
+declare module 'react' {
+  interface Attributes {
+    sx?: ThemeUIStyleObject
+  }
+}
+```
+
+### Using next.js
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    [
+      'next/babel',
+      {
+        'preset-react': {
+          importSource: 'theme-ui', // or '@theme-ui/core'
+          runtime: 'automatic',
+          throwIfNamespace: false,
+        },
+      },
+    ],
+  ],
+  // ...
+}
+```
```

Comment:
````suggestion
```

You can read more about [customizing Babel config in Next.js docs](https://nextjs.org/docs/advanced-features/customizing-babel-config).
````